### ISSUE TITLE
childId parameter missing in twig when baseRoutePattern attribute is set

### ISSUE THREAD

**Issue by Andreeey** [sonata-project/SonataAdminBundle#2651]
Hi guys!

I am trying to make admin url's more friendly with baseRoutePattern attribute http://sonata-project.org/bundles/admin/master/doc/reference/routing.html#route-patterns-urls. It works ok with a single entity but it fails with an exception when I go to related entity list. I have country and countrytranslate entities if i set baseRoutePattern for both country works ok on a new url but countrytranslate fails with this exception:

URL = /someprefix/countrytranslate/list
EXCEPTION = An exception has been thrown during the rendering of a template ("Some mandatory parameters are missing ("childId") to generate a URL for route "admin_vendor_bundle_country_countrytranslate_show".") in SonataAdminBundle:CRUD:list__action_show.html.twig at line 13. 

When I remove baseRoutePattern everything works fine.
services.yml:

```
vendor_bundle_admin.admin.country:
    class: Vendor\Bundle\AdminBundle\Admin\CountryAdmin
    calls:
        - [ addChild, ["@vendor_bundle_admin.admin.countrytranslate"]]
        - [ setTranslationDomain, [VendorAdminBundle]]
    tags:
        - {name: sonata.admin, manager_type: orm, group: Countries, label: Country}
    arguments:
        - ~
        - Vendor\Bundle\AdminBundle\Entity\Country
        - ~
vendor_bundle_admin.admin.countrytranslate:
    class: Vendor\Bundle\AdminBundle\Admin\CountryTranslateAdmin
    calls:
        - [ setTranslationDomain, [VendorAdminBundle]]
    tags:
        - { name: sonata.admin, manager_type: orm, group: "Countries", label: "Country translate" }
    arguments:
        - ~
        - Vendor\Bundle\AdminBundle\Entity\CountryTranslate
        - ~
```

I can also provide CountryAdmin and CountryTranslateAdmin classes but there is nothing extraordinary. 
CountryAdmin has attribute:
    protected $baseRoutePattern = 'country';

CountryTranslateAdmin has attributes:
    protected $baseRoutePattern = 'countrytranslate';
    protected $parentAssociationMapping = 'country';

**Comment by nilov** [comment]
I had the same problem. This pattern "$baseRoutePattern = 'countrytranslate'" for admin class which is a child is not correct. The easiest option is not to touch this property :) If you still need to override the pattern, it should be: "$baseRoutePattern = 'country/{id}/countrytranslate'"

**Comment by lushc** [comment]
I came across this issue while upgrading an existing project from 2.2 due to the changes in b35581139a0e110adc11aad20258339b983746bc. You'll encounter this error when the admin is a child and its route pattern doesn't follow the expected format of having both `id`  and `childId` request attributes present, e.g. editing a customer's order: `/customer/{id}/orders/{childId}/edit`. Since I had base route patterns without the `id` parameter, I ended up overriding the admin's getRequest method to ensure the request attribute is always set:

```php
public function getRequest()
{
    if ($this->isChild() && $this->hasRequest() && !$this->request->attributes->has($this->getParent()->getIdParameter())) {
        $this->request->attributes->set($this->getParent()->getIdParameter(), null);
    }

    return parent::getRequest();
}
```
